src/nvim/eval/typval_encode.c.h

```diff
@@ -351,8 +351,16 @@ static int TYPVAL_ENCODE_CONVERT_ONE_VALUE(
     break;
   case VAR_PARTIAL: {
     partial_T *const pt = tv->vval.v_partial;
-    (void)pt;
-    TYPVAL_ENCODE_CONV_FUNC_START(tv, (pt == NULL ? NULL : partial_name(pt)));
+    char *const fname = pt == NULL ? NULL : partial_name(pt);
+    char *ffname = NULL;
+    if (fname != NULL && pt->pt_name == NULL && isupper(fname[0])) {
+      size_t len = strlen(fname) + 3;
+      ffname = xmalloc(len);
```

Comment:
Can't find the maximum length for Vim function name to instead try a static buffer.